lib/AuthenticationSDK/util/MLEUtility.rb

```diff
@@ -0,0 +1,125 @@
+          raise StandardError.new('Failed to get certificate for MLE')
+        end
+        certificate = OpenSSL::X509::Certificate.new(Base64.decode64(cert_der))
+        validate_certificate(certificate, merchant_config.mleKeyAlias, @log_obj)
+        serial_number = extract_serial_number_from_certificate(certificate)
+        if serial_number.nil?
+          @log_obj.logger.error('Serial number not found in certificate for MLE')
+          raise StandardError.new('Serial number not found in MLE certificate')
+        end
+
+        jwk = JOSE::JWK.from_key(certificate.public_key)
+        if jwk.nil?
+          @log_obj.logger.error('Failed to create JWK object from public key')
+          raise StandardError.new('Failed to create JWK object from public key')
+        end
+        headers = {
+          'alg' => 'RSA-OAEP-256',
+          'enc'  => 'A256GCM',
+          'typ' => 'JWT',
+          'kid' => serial_number,
+          'iat' => Time.now.to_i
+        }
+        jwe = JOSE::JWE.block_encrypt(jwk, request_payload, headers)
+
+        compact_jwe = jwe.compact
+        mle_request_body = create_request_payload(compact_jwe)
+        @log_obj.logger.debug('LOG_REQUEST_AFTER_MLE: ' + mle_request_body)
+        return mle_request_body
+      rescue StandardError => e
+        @log_obj.logger.error("An error occurred during encryption: #{e.message}")
+        raise e
+      end
+    end
+
+
+    def self.get_certificate(merchant_config, log_obj)
+      begin
+        p12_file_path = File.join(merchant_config.keysDirectory, merchant_config.keyFilename + '.p12')
+        file = File.binread(p12_file_path)
+        p12_file = OpenSSL::PKCS12.new(file, merchant_config.keyPass)
+        x5_cert_pem = OpenSSL::X509::Certificate.new(p12_file.certificate)
+        x5_cert_pem.subject.to_a.each do |attribute|
+          return x5_cert_pem if attribute[1].include?(merchant_config.mleKeyAlias)
+        end
+        p12_file.ca_certs.each do |cert|
+          cert.subject.to_a.each do |attribute|
+            return cert if attribute[1].include?(merchant_config.mleKeyAlias)
+          end
+        end
+      rescue OpenSSL::PKCS12::PKCS12Error => e
+        log_obj.logger.error("Failed to load PKCS12 file: #{e.message}")
+        raise e
+      rescue OpenSSL::X509::CertificateError => e
+        log_obj.logger.error("Failed to create X509 certificate: #{e.message}")
+        raise e
+      rescue StandardError => e
+        log_obj.logger.error("An error occurred while getting the certificate: #{e.message}")
+        raise e
+      end
+    end
+
+    def self.validate_certificate(certificate, mle_key_alias, log_obj)
+      if certificate.not_after.nil?
+        log_obj.logger.warn("Certificate for MLE don't have expiry date.")
+      end
+      if certificate.not_after < Time.now
+        log_obj.logger.warn('Certificate with MLE alias ' + mle_key_alias + ' is expired as of ' + certificate.not_after.to_s + ". Please update p12 file.")
```

Comment:
if certificate is expiry. Please throw error here and verify that user is getting error here. 
Once verified then mark the throw error as commented line. Currently we support expiry certificate for mle but will enable this feature in future.